### ISSUE TITLE
show_stat/show_all show the whole class docstring

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#545]
The output of `show_stat` (and `show_all`) displays the statistic "name" and then the whole docstring for the `sherpa.stats` class:

```
In [2]: sherpa.astro.ui.show_stat()                                                          
Statistic: Chi2Gehrels
Chi Squared with Gehrels variance.

    The variance is estimated from the number of counts in each bin,
    but unlike `Chi2DataVar`, the Gaussian approximation is not
    used. This makes it more-suitable for use with low-count data.

... and a lot more ...
```

Before I started adding/enhancing docstrings to everything, the `sherpa.stats` classes used to just have a one-line docstring, so the output used to look like:

 ```
```
In [2]: sherpa.astro.ui.show_stat()                                                          
Statistic: Chi2Gehrels
Chi Squared with Gehrels variance.

In [3]:
```

We could do one of:

1. extract just the first line from the docstring
2. add a method to get the "summary text" for the statistic (which is a more formalized way of step 1)
3. rethink what we want the output of `show_stat` to be (e.g. compare to `show_method`)

**Comment by hamogu** [comment]
Why do you use `show_fit` in loops? There might be a better way of getting the information that you actually are looking for.

**Comment by leopfd** [comment]
Right, I generally don't use `show_fit` in loops because it's not the best way to extract specific results. However, I find it useful for getting quick, readable summaries. That said, even a single call to `show_fit` can produce an overly verbose output, which is why a reduction to one line would still be very helpful.

**Comment by hamogu** [comment]
Maybe it's easier to show than to describe. I'll be in the office tomorrow and would like to have a look, if possible?
It's always hard to understand what users are actually looking for and thus to design the user interface in ways that work well and we don't effete get feedback like this, so much of the UI is designed with "users in mind", but based on the personal biases of very few developers.
(I can't find the link now, but at some point the developers of Visual Studio at Microsoft spend dozens of hours watching developers code in Visual studio through a half-transparent mirror because they wanted know which buttons people click and where they get stuck without constantly having someone sit right next to you.)

**Comment by hamogu** [comment]
Instead of using `show_fit`, you can just get the fit result and print that to the screen:
```Python
>>> fres = ui.get_fit_results()
>>> print(fres)
datasets       = (1,)
itermethodname = none
methodname     = levmar
statname       = cstat
succeeded      = True
parnames       = ('bkg.c0', 'line.ampl')
parvals        = (2.4444414372811707, -15.0)
statval        = 4.859179942147372
istatval       = 103.30091057882433
dstatval       = 98.44173063667695
numpoints      = 10
dof            = 8
qval           = 0.7725157254781938
rstat          = 0.6073974927684215
message        = successful termination
nfev           = 16
>>> fmod = ui.get_model()
>>> print(fmod)
const1d.bkg + delta1d.line
   Param        Type          Value          Min          Max      Units
   -----        ----          -----          ---          ---      -----
   bkg.c0       thawed      2.44444 -3.40282e+38  3.40282e+38           
   line.pos     frozen            5 -3.40282e+38  3.40282e+38           
   line.ampl    thawed          -15 -3.40282e+38  3.40282e+38   
```

or to a file:
```Python
>>> with open('output.txt', 'w') as f:
...     f.write(str(fmod))
```

@leopfd : Would that solve your use case?

**Comment by DougBurke** [comment]
So, a "meta issue" here is that there are ways to get at the information that @leopfd (and others) want without going through `show_fit`, but they just don't know about it. I've tried to make sure that there are "See Also" sections, but as noted for `show_fit` is may not be useful (as they are more about "other show methods" than "general data access")

- https://cxc.cfa.harvard.edu/sherpa/ahelp/show_fit.html#seealso
- https://sherpa.readthedocs.io/en/4.17.0/ui/api/sherpa.astro.ui.show_fit.html

Hmm. In this case the "RTD" version is probably more useful than the ahelp one, which I can look at (as I generate the ahelp version from the Python docstring).

However, there's always the issue that something like "See Also" is never going to help all people, so how else do we document to users useful features?

For instance, maybe you want [`ui.calc_stat`](https://sherpa.readthedocs.io/en/4.17.0/ui/api/sherpa.astro.ui.calc_stat.html) or [`ui.calc_stat_info`](https://sherpa.readthedocs.io/en/4.17.0/ui/api/sherpa.astro.ui.calc_stat_info.html) as well as, or instead of `ui.get_model`. Or you want to do `print(ui.get_model().thawedpars)` or ... 

This is basically saying

- `show_fit` is probably not what is wanted here [*]
- there are other ways to get the data
- but how do we tell the user

[*] that's not to say that `show_fit` output can not be simplified, but probably not by as much as @leopfd would want

**Comment by leopfd** [comment]
@hamogu I’d say this is a perfectly reasonable way to quickly display a result. Any other concerns would mostly come down to personal preference — and as you saw, I was able to work around it by building my own summary from `get_fit_results`.

Just for context: as a relatively new user, a few years in, I originally picked up `show_fit` from the [Introduction to PHA spectra](https://cxc.cfa.harvard.edu/sherpa/threads/pha_intro/index.html) help thread. I referred to it often early on, so I saw `show_fit` as the go-to way to summarize fit results, and `get_fit_results` as the more programmatic option for accessing specific values.

My only suggestion for `show_fit` would be to simplify the docstring output so the two main blocks of information aren’t separated by a wall of text. But as discussed, it’s not a huge issue.

@DougBurke I often find myself scrolling to the 'See Also' section when reading the ahelp pages, but if there’s not much listed, I end up scanning the index in the sidebar. A reference page with all commands that groups related commands by topic or functionality could be very helpful.
